### PR TITLE
Added combined servlet to handle Spoonacular requests

### PR DESCRIPTION
**Pull request by joeychin01**
Added SpoonacularCombinedServlet.java to handle all spoonacular code in one file. I kept the other two servlets just in case they might be useful in the future, but for the prototype the combined servlet should handle most of the requests. Updated recipe.html to use the combined servlet, and made minor bug and style fixes

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/SpoonacularCombinedServlet.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String query = request.getParameter("dishName");
+    Client client = ClientBuilder.newClient();
+    WebTarget target = client.target(spoonacularPrefix + "/search?query=" + query + "&number=2&apiKey=" + spoonacularAPIKey);
+    try {
+      String recipeListJSONString = target.request(MediaType.APPLICATION_JSON).get(String.class);
+      JSONObject recipeJson = new JSONObject(recipeListJSONString);
+      JSONArray recipeListJson = new JSONArray(recipeJson.get("results").toString());
+      String recipeList = "";
+      int i = 0;
```
Comment (ALai2) [inline src/main/java/com/google/sps/servlets/SpoonacularCombinedServlet.java:47]: nit: maybe change i to another name, like "isFirstInList"?

File: src/main/java/com/google/sps/servlets/SpoonacularCombinedServlet.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+/** Servlet to take in dish name and return bulk recipe infomation */
+@WebServlet("/recipeInfo")
+public class SpoonacularCombinedServlet extends HttpServlet {
+  static String spoonacularPrefix = "https://api.spoonacular.com/recipes";
+  static String spoonacularAPIKey = "cd2269d31cb94065ad1e73ce292374a5";
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String query = request.getParameter("dishName");
+    Client client = ClientBuilder.newClient();
+    WebTarget target = client.target(spoonacularPrefix + "/search?query=" + query + "&number=2&apiKey=" + spoonacularAPIKey);
```
Comment (liguanghua) [inline src/main/java/com/google/sps/servlets/SpoonacularCombinedServlet.java:41]: Consider String.format()?

File: src/main/java/com/google/sps/servlets/SpoonacularCombinedServlet.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    try {
+      String recipeListJSONString = target.request(MediaType.APPLICATION_JSON).get(String.class);
+      JSONObject recipeJson = new JSONObject(recipeListJSONString);
+      JSONArray recipeListJson = new JSONArray(recipeJson.get("results").toString());
+      String recipeList = "";
```
Comment (liguanghua) [inline src/main/java/com/google/sps/servlets/SpoonacularCombinedServlet.java:46]: Consider String.join() (https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#join-java.lang.CharSequence-java.lang.Iterable-) or Collectors.joining() (https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collectors.html#joining-java.lang.CharSequence-)

File: src/main/webapp/script.js [hunk 4]
```diff
@@ -115,14 +115,23 @@ function clearSavedProfileStatus() {
   profileStatusElement.style.display = "none";
 }
 
-/** Gets recipe id list from query string */
+
+function getRecipe(){
+  /** Function gets recipe information from user input ID and displays the title on the page */
+  var numRecipe = document.getElementById("num-recipe").value;
+  fetch('/recipeInfo?numRecipe='+numRecipe).then(response => response.json()).then((recipeInfo) => {
```
Comment (liguanghua) [inline src/main/webapp/script.js:122]: Are the enclosing parenthesis around "recipeInfo" necessary?

Comment (joeychin01) [reply src/main/webapp/script.js:122]: I believe so

**Pull request merged**